modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -743,71 +743,90 @@ private boolean isExpressionFunction(MetadataType metadataType)
         }
     }
 
-    private ValueResolver parseDateTime(Object value, DateTimeType dataType, Object defaultValue)
+
+    private static ValueResolver doParseDate(Object value, Class<?> type)
     {
-        Class<?> type = getType(dataType);
-        if (isExpression(value, parser))
-        {
-            return new TypeSafeExpressionValueResolver((String) value, type);
-        }
 
-        if (type.equals(LocalDateTime.class))
+        if (value instanceof String)
         {
-            Date date = doParseDate(value, DATE_FORMAT, defaultValue);
-            return new StaticValueResolver(LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()));
+            Object constructedValue = null;
+            DateTime dateTime = getParsedDateTime((String) value);
+
+            if (type.equals(LocalDate.class))
+            {
+                constructedValue = LocalDate.of(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
```

Comment:
No. That is wrong. Date class can handle time. Calendar class can handle time and timeZone as well. The only real limitant that you have here is that of the framework you want to use to parse the String. If Joda doesn't do what you want then drop Joda, don't change the feature to match the code you have. I'm sure the internet is full of examples of how to do that.

What we need is to be able to parse ANY iso String into ANY of the aforementioned types.